### PR TITLE
Add flood fill

### PR DESCRIPTION
**Pull request by algobytewise**
Please include a summary of the change.
Please also include relevant motivation and context (if applicable).
Put 'x' in between square brackets to mark an item as complete.

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Comments in areas I changed are up to date
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have made corresponding changes to the README.md
- [x] I have made corresponding changes to the DIRECTORY.md

### CODE REVIEW

**Comment by algobytewise** [comment]
I'm not sure what codecov/project is complaining about since all the public methods have tests for their exceptions and their regular behavior. The report page (https://codecov.io/gh/TheAlgorithms/C-Sharp/pull/224?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=None) results in an error message.

File: Algorithms/Other/FloodFill.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    /// Flood fill, also called seed fill, is an algorithm that determines and
+    /// alters the area connected to a given node in a multi-dimensional array with
+    /// some matching attribute. It is used in the "bucket" fill tool of paint
+    /// programs to fill connected, similarly-colored areas with a different color.
+    /// (description adapted from https://en.wikipedia.org/wiki/Flood_fill)
+    /// (see also: https://www.techiedelight.com/flood-fill-algorithm/).
+    /// </summary>
+    public static class FloodFill
+    {
+        private static List<ValueTuple<int, int>> neighbors = new List<ValueTuple<int, int>> { (-1, -1), (-1, 0), (-1, 1), (0, -1), (0, 1), (1, -1), (1, 0), (1, 1) };
```
Comment (siriak) [inline Algorithms/Other/FloodFill.cs:18]: You can use List<(int xOffset, int yOffset)>

**Review by siriak — COMMENTED** [review]
Why are documentation files changed so much? Please check what could lead to this and make only necessary changes.

File: Algorithms/Other/FloodFill.cs (hunk 1 above)
Comment (algobytewise) [reply Algorithms/Other/FloodFill.cs]: Not sure what happened with the documentation files, maybe some quirk with my editor. I'm still getting used to value tuples, they seem to be quite convenient.

**Review by siriak — APPROVED** [review]
Sorry for the late response, I was on a small holiday. Looks good!

**Pull request merged**